sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -785,7 +785,23 @@ public User getUser() {
   private String getAndroidId() {
     // Android 29 has changed and -> Avoid using hardware identifiers, find another way in the
     // future
-    return Settings.Secure.getString(context.getContentResolver(), Settings.Secure.ANDROID_ID);
+    String androidId =
+        Settings.Secure.getString(context.getContentResolver(), Settings.Secure.ANDROID_ID);
+
+    //    https://android-developers.googleblog.com/2011/03/identifying-app-installations.html
+    if (androidId == null
+        || androidId.isEmpty()
+        || androidId.toLowerCase(Locale.ROOT).contentEquals("9774d56d682e549c")) {
+      try {
+        androidId = Installation.id(context);
+      } catch (RuntimeException e) {
+        log(SentryLevel.ERROR, "Could not generate device Id.", e);
+
+        androidId = UUID.randomUUID().toString();
```

Comment:
Would it make sense to return `null` here instead? We'll potentially artificially increase the cardinality of the device id without any added benefit to the client. I'll look like there're more users then in fact are and cost us more to process the event.

Reply:
yeah agreed, just thought it would make sense to have a sort of ID